tests/Admin/AdminTest.php

```diff
@@ -2298,8 +2298,12 @@ public function testDefaultFilters(): void
             ],
         ], $admin->getFilterParameters());
 
+        // NEXT_MAJOR: Remove next asserts
```

Comment:
I think you need to change
```
public function testDefaultFilters(): void
```
To
```
    /**
     * NEXT_MAJOR: Remove the assertion about isDefaultFilter method. 
     *
     * @legacy
     *
     * @expectDeprecation Method "Sonata\AdminBundle\Admin\AbstractAdmin::isDefaultFilter" is deprecated since sonata-project/admin-bundle 3.x.
     */
    public function testDefaultFilters(): void
```

You will get rid of the test failure.

`$this->expectDeprecation();` is not effective.